src/indexers/hosted/indexer.ts

```diff
@@ -65,6 +66,8 @@ export type IndexerConfig = {
   schedule: IndexerSchedule;
 };
 
+export type RunOptions = { waitForCompletionTimeoutMs?: number };
```

Comment:
What is the rationale for a timeout vs just a simple boolean?

Reply:
I started with a Boolean AND a timeout - we definitely need a timeout because deciding how long to wait is up to the user, even if we provide a default. Then I realized that we can use the presence of a timeout to obviate the need for a Boolean, we just need one knob instead of two

Reply:
I guess we'll also need to document that an Error will be thrown if timeout has been exceeded without indexer completing - but that's even more true if we provide a default timeout implicitly

Reply:
Why can't the default timeout be no timeout?

Reply:
Meaning we wait indefinitely long?

Reply:
I instinctively didn't even consider this option because I've been conditioned on providing a timeout for any waiting API, but given huge variation in indexer execution time, this may be a good option!

Is -1 also considered a standard for "infinite" in TS world?

Reply:
For not let's not even bother worrying about timeouts until someone complains about it. Let's assume we don't timeout. We can add a timeout parameter in the future if needed, but let's avoid the additional complexity for now.

Reply:
The user can definitely make a timeout themselves a number of ways (eg Promise.race).

Reply:
This is now exactly in the shape we discussed, afaict. Going to merge this tomorrow morning unless I get more feedback.

Reply:
(reminder we need to turn on approval requirements for this repo and all others)

Reply:
Will turn them on for this and website repo. We've had it on for the cortex repo for a while.